kpet/data.py

```diff
@@ -213,6 +213,8 @@ def __init__(self, data):
                     match=Class(PositivePattern),
                     dont_match=Class(NegativePattern),
                     waived=Boolean(),
+                    role=String(),
```

Comment:
Another thing: please assign the default `role` right here in the code, so you don't have to do it in the template and can just assume it always exists.